multiproof.go

```diff
@@ -244,7 +255,14 @@ func (mp *MultiProof) Read(r io.Reader) error {
 		return fmt.Errorf("failed to read D: %w", err)
 	}
 	mp.D = *D
-	mp.IPA.Read(r)
+	if err := mp.IPA.Read(r); err != nil {
+		return fmt.Errorf("failed to read IPA proof: %w", err)
+	}
+	// Check that the next read is EOF.
+	var buf [1]byte
+	if _, err := r.Read(buf[:]); err != io.EOF {
+		return errors.New("expected EOF")
+	}
```

Comment:
This is another situation discovered by the fuzzer, but actually I think it's pretty subjective so open to comments.

The situation is that you can add more bytes at the tail of the serialized bytestream and they will be ignored since reading from the `io.Reader` stops at the expected byte length. Any further bytes are ignored and "left" on the Reader (if any).

Here I'm taking a step further and do an extra read of 1 byte and expect the reader to give an EOF. I think this is safer if in the future there's some new deserialization of proofs with old data that might contain those extra "unread" bytes and maybe produce an unfortunate situation? Maybe being too paranoid, but I'm not sure there's a good reason to allow "extra bytes" here, so happy to chat about it!